api4/bot.go

```diff
@@ -147,6 +148,45 @@ func getBots(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write(bots.ToJson())
 }
 
+func getBotByName(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireBotUserName()
+	if c.Err != nil {
+		return
+	}
+	botUserName := c.Params.BotUserName
+
+	includeDeleted := r.URL.Query().Get("include_deleted") == "true"
+
+	bot, err := c.App.GetBotByName(botUserName, includeDeleted)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	if c.App.SessionHasPermissionTo(c.App.Session, model.PERMISSION_READ_OTHERS_BOTS) {
```

Comment:
1/5 more idiomatic and simpler would be (if I got it right?)
```
if !c.App.SessionHasPermissionTo(c.App.Session, model.PERMISSION_READ_OTHERS_BOTS) {
	if bot.OwnerId != c.App.Session.UserId ||
		!c.App.SessionHasPermissionTo(c.App.Session, model.PERMISSION_READ_BOTS) {

		// Pretend like the bot doesn't exist at all to avoid revealing that the
		// user is a bot.
		c.Err = model.MakeBotNotFoundError(botUserName)
		return
	}
}
```

Reply:
this is the same code as the `GetBot` @levb

Reply:
Right. @crspeller did you or someone else write that (`GetBot`)? Is my suggestion correct? (perhaps a as a separate PR)?